### DISCUSSION TITLE
preventing-breaking-changes-in-public-apis-with-publicapigenerator/

### DISCUSSION THREAD

**Discussion by giscus** [andrewlock/blog-comments#444] [Announcements]
# preventing-breaking-changes-in-public-apis-with-publicapigenerator/

In this post I show how you can use the PublicApiGenerator project to document your public API and build a snapshot unit test to prevent breaking changes

https://andrewlock.net/preventing-breaking-changes-in-public-apis-with-publicapigenerator/

**Comment by bitbonk** [comment]
There seems to be a somewhat new kid on the block that can help with preventing breaking changes in packages or assemblies and it is part of the .NET SDK: https://learn.microsoft.com/dotnet/fundamentals/apicompat/overview
You can use `Microsoft.DotNet.ApiCompat.Tool` or the msbuild tasks with `<ApiCompatValidateAssemblies>true</ApiCompatValidateAssemblies>` or `<EnablePackageValidation>true</EnablePackageValidation>` to do [all sort of](https://learn.microsoft.com/en-us/dotnet/fundamentals/apicompat/diagnostic-ids) compatibility validations.
